### DISCUSSION TITLE
Join data

### DISCUSSION THREAD

**Discussion by LuisOlivaresJ** [jrkerns/quaac#3] [Ideas]
On Documentation, does it make sense to create another section named "Join data" or something like that?

I was trying to figure out how the workflow is with QuAAC when we have subsequent measurements. Here are some ideas that could be useful to include as examples:

1. Use of Document.merge() method.
2. Get a user and/or equipment from a previously created yaml (json) file. (I am a little lazy at typing, and I don't want to create another User instance if I miss a single character), or should it be the responsibility of the client or a third party application?
3. A remainder that a Document could have any number of DataPoints (not necessary for the same kind of measurement). 
4. An example of how to filter using DataPoint.name attribute and plot the measurement tendency with matplotlib.

Please let me know what you think (or if I misunderstood something). Additionally, I can create a new reStructuredText file with the previous ideas and make a pull request to include it if you want.

**Comment by jrkerns** [comment]
Hey Luis, 
     Good ideas. Documenting the merge method is probably worthwhile. For 2) I actually think this might work well when I get to supporting SQLite. At that point one could write SQL or use an ORM like [peewee](https://docs.peewee-orm.com/en/latest/index.html) which support retrieval and filtering out of the box. 3) sure. 4) sure.

**Comment by LuisOlivaresJ** [comment]
Added in #4. 
Please let me know I you need help with something. I'll be happy to collaborate.

**Reply by jrkerns** [reply]
I'm not a very good maintainer unfortunately so sorry for the slow responses, but if you're willing to take some images on a linac it would help validate some of the pylinac new features (separate topic from quaac).

**Reply by LuisOlivaresJ** [reply]
Sure. I am going to start with [Picket Fence](https://pylinac.readthedocs.io/en/latest/picketfence.html) (actually, I was trying it in the morning to see how it works).

**Reply by LuisOlivaresJ** [reply]
PicketFence.to_quaac() method **works as expected**. [Here](https://github.com/LuisOlivaresJ/quaac/blob/jupyter_example_PF/notebook_examples/PF_QuAAC_Luis.ipynb) you can see an example usage in a notebook.

By my understanding, after ```to_quaac()``` is called, it is not easy to discriminate DataPoints by parameters of adquisition (for example, angle gantry).  As can be seen in Out[25], the ```parameters``` field only has "pylinac version".

It seems that QuaacDatum class should have an extra field
https://github.com/jrkerns/pylinac/blob/f65e73e2df0ef652256c33682f6fd17faa0d0532/pylinac/core/utilities.py#L266-L274
or a change to ```QuaacMixin.to_quaac()``` for an extra parameter to allow the user to include parameters
https://github.com/jrkerns/pylinac/blob/f65e73e2df0ef652256c33682f6fd17faa0d0532/pylinac/core/utilities.py#L285-L296

Please let me know if I am missing something.
